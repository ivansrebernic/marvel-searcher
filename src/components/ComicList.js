import React from 'react'
import styled from 'styled-components'

const ComicListStyled = styled.ul`
    background-color:#F7F8FA;
    margin:0;
    padding:0;
    max-width:500px;
    height:100px;
    width:100%;
    overflow:scroll;
    list-style-type:none;
    overflow-x:hidden;
`
const ComicItem = styled.li`
    display:flex;
    justify-content:space-between;
    border-bottom: 1px solid grey;
`


function ComicList(props) {



    return (
        <ComicListStyled>
            {
                props.comicList.map(comic => (
                    <ComicItem key={comic.name}>
                        <span>{comic.name}</span>
                        <span>[Fecha]</span>
                    </ComicItem>
                ))
            }
        </ComicListStyled >
    )
}

export default ComicList