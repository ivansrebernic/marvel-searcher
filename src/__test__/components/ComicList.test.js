import React from 'react'
import { shallow } from 'enzyme'
import ComicList from '../../components/ComicList'


describe('<ComicList>', () => {
    let wrapper
    beforeEach(() => {
        wrapper = shallow(<ComicList></ComicList>)
    })
    it('Renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    });
    it('Fetches comics', () => {
        const spy = jest.spyOn(global, 'fetch')
        setTimeout(() => {
            wrapper.update()
            expect(spy).toHaveBeenCalledTimes(1)
        })
    });
    it('displays a set of comics', () => {
        const comics = [{ id: 1, title: "comic title 1" }, { id: 2, title: "comic title 2" }, { id: 3, title: "comic title 3" }]
        wrapper.setState({ comics: comics });
        console.log(wrapper.debug())
        expect(wrapper.find('span')).toHaveLength(3)
    })
})