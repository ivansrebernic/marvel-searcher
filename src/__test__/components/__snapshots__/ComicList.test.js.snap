// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicList> Renders correctly 1`] = `
<styled.ul
  onScroll={[Function]}
>
  <li>
    There is no comics registered from this super-hero
  </li>
</styled.ul>
`;
